service.go

```diff
@@ -0,0 +1,337 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"strings"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	apiError "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/selection"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/tools/cache"
+)
+
+// Struct to build service watcher which looks after target services.
+type ServiceWatcher struct {
```

Comment:
Should this be renamed to `Watcher`? What if we start watching endpointslices?

Reply:
So in EndpointSlices, I have done this : Its kind of parent struct 
https://github.com/rushi47/service-mirror-prototype/pull/10/files#diff-0ea94270ccbc3fa46cbb5b0c86ea055db059190c02ca103a32426192f6ca3d78R27-R33

Which is extended by EndpointWatcher and ServiceWatcher
svc watcher : https://github.com/rushi47/service-mirror-prototype/pull/10/files#diff-c52daedaeda258db15ec11d57bbf299412a20328f6cd026980ef07f01960dd3aR22-R24

ep watcher : https://github.com/rushi47/service-mirror-prototype/pull/10/files#diff-92d7790692abf52c09d20d29b7945730bed18aac6643dbc4488a2b530e9390d7R20-R24

ps : This solves kind of problem of logging but not in straightforward way. I will change with what you suggested above. But to main your main question i hope this helps.